### PR TITLE
Update examples

### PR DESCRIPTION
**Pull request by nwu63**
## Purpose
This PR updates some of the examples that were deprecated. It also changes the type of error raised when the SqliteDict is not found.

## Type of change
What types of change is it?
_Select the appropriate type(s) that describe this PR_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (non-backwards-compatible fix or feature)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Documentation update
- [x] Maintenance update
- [ ] Other (please describe)

## Testing
All the examples work now, especially when using non-default command line arguments.

## Checklist
_Put an `x` in the boxes that apply._

- [x] I have run `flake8` and `black` to make sure the code adheres to PEP-8 and is consistently formatted
- [x] I have run unit and regression tests which pass locally with my changes
- [ ] I have added new tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation

### CODE REVIEW

**Review by Xiaosong2105 — COMMENTED** [review]
The updated way of constructing funcsSens dictionary looks good.
The pyoptsparse.History and sqlitedict.SqliteDict are basically doing the same thing, correct?

File: examples/rosenbrock.py [hunk 1]
```diff
@@ -45,87 +32,50 @@ def sensfunc(xdict, funcsDict):
...
-        opt.setOption("Major iterations limit", 10)
-        solSnopt1 = opt(optProb, sens=sens, sensMode="pgc", storeHistory="opt_hist")
 
-        # Now we are allowed to do 50
-        opt.setOption("Major iterations limit", 50)
-        if testHist == "hot":
-            solSnopt2 = opt(optProb, sens=sens, sensMode=sensMode, hotStart="opt_hist", storeHistory="opt_hist")
-        else:
-            solSnopt2 = opt(optProb, sens=sens, sensMode=sensMode, coldStart="opt_hist", storeHistory="opt_hist")
+if args.testHotstart:
```
Comment (marcomangano) [inline examples/rosenbrock.py:50]: Is this the best practice to map a specific option to different optimizers? What if we are dealing with multiple options and not just `max_iter`?

**Review by marcomangano — COMMENTED** [review]
See below, looks good for the rest.

**Comment by nwu63** [comment]
> The pyoptsparse.History and sqlitedict.SqliteDict are basically doing the same thing, correct?
Almost. The `History` class is basically an instance of the SqliteDict, but with its specific format.

File: examples/rosenbrock.py (hunk 1 above)
Comment (nwu63) [reply examples/rosenbrock.py:50]: Yeah this is a good point. Maybe we should restrict this to run with SLSQP only, and then everything would be simpler.

**Review by Xiaosong2105 — APPROVED** [review]

File: examples/rosenbrock.py (hunk 1 above)
Comment (nwu63) [reply examples/rosenbrock.py:49]: I've updated the rosenbrock example to only run SLSQP.

**Review by marcomangano — APPROVED** [review]

**Pull request merged**